scio-test/src/test/scala/com/spotify/scio/io/FileFormatTest.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2017 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.io
+
+import java.nio.file.Files
+
+import com.google.protobuf.Timestamp
+import com.spotify.scio.ScioContext
+import com.spotify.scio.testing.PipelineSpec
+
+class FileFormatTest extends PipelineSpec {
+
+  // Legacy files generated with 0.3.0-beta1, the last release before the
+  // BufferedPrefix{Input,Output}Stream change.
+
+  private val path = "scio-test/src/test/resources"
+
+  private val objects = (1 to 100).map(x => (x, x.toDouble, x % 2 == 0, "s" + x))
+  private val protobufs = (1 to 100)
+    .map(x => com.google.protobuf.Timestamp.newBuilder.setSeconds(x * 1000).setNanos(x).build)
+
+  // Object file is NOT backwards compatible
```

Comment:
In fact I got `KryoException: Unable to find class` error when reading an object file written with 0.3.0-beta1. Need to investigate if it's fixable.